wicket-core/src/main/java/org/apache/wicket/page/PageAccessSynchronizer.java

```diff
@@ -93,8 +94,8 @@ public Duration getTimeout(int pageId)
 	 */
 	public void lockPage(int pageId) throws CouldNotLockPageException
 	{
-		final Thread thread = Thread.currentThread();
-		final PageLock lock = new PageLock(pageId, thread);
+		final RequestCycle cycle = RequestCycle.get();

```

Comment:
In that timeframe the page should be unlocked. On resume the new thread should acquire a new lock or wait if it is not available at the moment.